packaging/docker/docker-entrypoint

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Add default apm-server arguments.
+APM_SERVER="apm-server --environment=container"
+
+# If no arguments are provided, just run "apm-server".
+if [[ $# -eq 0 ]]; then
+  exec $APM_SERVER
+fi
+
+# Check if the the user has invoked the image with flags.
+# eg. "apm-server -c apm-server.yml"
+if [[ -z $1 ]] || [[ ${1:0:1} == '-' ]] ; then
+  exec $APM_SERVER "$@"
+else
+  # The user may be looking for a subcommand, like "apm-server apikey".
+  subcommands=$(apm-server help \
+                  | awk 'BEGIN {RS=""; FS="\n"} /Available Commands:/' \
+                  | awk '/^ / {print $1}')
+
+  # If we _did_ get a subcommand, pass it to apm-server.
+  for subcommand in $subcommands; do
+      if [[ $1 == $subcommand ]]; then
+        exec $APM_SERVER "$@"
+      fi
+  done
+fi
+
+# If neither of those worked, then they have specified the binary they want, so
+# just do exactly as they say.
```

Comment:
Two questions, What does "they have specified the binary they want" mean? and Under what circumstances will this case be useful?

Reply:
FYI this is copied from beats. This means the user has run something like `docker run apm-server-image bash` (run `bash` inside the container). The alternatives are `docker run apm-server-image -e` (run `apm-server -e` inside the container) and `docker run apm-server-image test output` (run `apm-server test output` in the container).

TBH I think this entrypoint is overcomplicated, and it might be better to just require users to specify the apm-server command like `apm-server run apm-server-image apm-server -e`. Nevertheless, it remains this way for backwards compatibility.

Reply:
Got it, thanks for the explanation.